node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/Preparer.java

```diff
@@ -92,10 +92,24 @@ public List<Node> prepare(ApplicationId application, ClusterSpec cluster, NodeSp
         }
     }
 
+    /**
+     * Whether the preparation of an application MAY require changes to the parent hosts, and therefore the parent host lock is required.
+     * See {@link NodeCandidate#withExclusiveParent(boolean)}.
+     */
+    public static boolean requireParentHostLock(boolean makeExclusive, NodeType type, boolean allowHostSharing) {
```

Comment:
Something like: `canMakeEmptyHostExclusive`, perhaps? I'm sure there are better variations than that, too.

Reply:
Good suggestion